.github/workflows/backend-deploy.yaml

```
@@ -78,9 +78,22 @@ jobs:
       DATABASE_URL_STAGING: ${{ secrets.DATABASE_URL_STAGING }}
       BACKEND_DEPLOY_STAGING_WEBHOOK: ${{ secrets.BACKEND_DEPLOY_STAGING_WEBHOOK }}
     steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install NodeJS
+        uses: actions/setup-node@v3
+        with:
+          node-version: "20"
+
+      - name: Install dependencies
+        working-directory: ./backend
+        run: npm ci
+
       - name: Run DB migration
         working-directory: backend
         run: npx -y dbmate -e DATABASE_URL_STAGING --no-dump-schema up
```

I think the -y flag might be unnecessary here since dbmate will already have been installed in the previous step.
```suggestion
        run: npx dbmate -e DATABASE_URL_STAGING --no-dump-schema up
```

---

Yeah, technically it is not needed if the package is installed; having it there won't cause any harm though. 
I'll remove the `-y`s from all the other places where it is not needed.